src/utils/compose_executor.py

```
@@ -0,0 +1,563 @@
+    def _compose_file_path(self) -> str:
+        return os.path.join(self._work_dir, COMPOSE_FILE_NAME)
+
+    def _compose_project_name(self, spec: workflow_module.WorkflowSpec) -> str:
+        sanitized = re.sub(r'[^a-z0-9-]', '-', spec.name.lower())
+        sanitized = re.sub(r'-{2,}', '-', sanitized).strip('-')
+        return f'osmo-{sanitized}' if sanitized else 'osmo-default'
+
+    def _compose_base_cmd(self, spec: workflow_module.WorkflowSpec) -> List[str]:
+        return (
+            self._compose_cmd.split()
+            + ['-p', self._compose_project_name(spec), '-f', self._compose_file_path]
```

Ruff (0.15.9)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)